File: backend/app/schemas/claim.py
from pydantic import BaseModel
from typing import List, Optional, Any
from datetime import datetime


class DocumentSchema(BaseModel):
    id: str
    claim_id: str
    document_type: str
    storage_url: str

    class Config:
        from_attributes = True


class ExtractedDataSchema(BaseModel):
    id: str
    claim_id: str
    extracted_json: Any

    class Config:
        from_attributes = True


class ClaimResponse(BaseModel):
    id: str
    status: str
    claimed_amount: float
    approved_amount: float
    confidence_score: float
    decision_reason: str
    created_at: datetime
    documents: List[DocumentSchema]
    extracted_data: Optional[ExtractedDataSchema] = None

    class Config:
        from_attributes = True


class DocumentInput(BaseModel):
    id: Optional[str] = None   # optional — generated server-side if omitted
    document_type: str
    storage_url: str


class ProcessClaimRequest(BaseModel):
    documents: List[DocumentInput]


class ManualAdjudicateRequest(BaseModel):
    status: str
    approved_amount: float
    comment: str


class PolicyConfigSchema(BaseModel):
    max_claim_amount: float
    high_value_manual_review_threshold: float = 25000.0
    min_claim_amount: float = 100.0
    min_confidence_threshold: float
    late_submission_days: int = 30
    doctor_reg_pattern: str = "^(?:[A-Z]{1,5}[0-9]{4,10}|[A-Z]{2}/[0-9]{1,10}/[0-9]{4})$"
    required_document_types: List[str] = ["Prescription"]
    non_covered_keywords: List[str]
    covered_keywords: List[str]

File: frontend/types/claim.ts
export interface Document {
  id: string
  claim_id: string
  document_type: string
  storage_url: string
}

export interface ExtractedJsonPayload {
  patient_name: string
  doctor_name: string
  doctor_registration: string
  diagnosis: string
  consultation_amount: number
  diagnostic_amount: number
  pharmacy_amount: number
  total_claim_amount: number
  treatment_date: string
  confidence_score: number
}

export interface ExtractedData {
  id: string
  claim_id: string
  extracted_json: ExtractedJsonPayload
}

export interface Claim {
  id: string
  status: 'APPROVED' | 'REJECTED' | 'PARTIAL_APPROVAL' | 'MANUAL_REVIEW' | string
  claimed_amount: number
  approved_amount: number
  confidence_score: number
  decision_reason: string
  created_at: string
  documents: Document[]
  extracted_data?: ExtractedData
}